File: src/main/scala/com/jxc454/udpstreaminganalytic/serializers/PbIntSerializer.scala
package com.jxc454.udpstreaminganalytic.serializers

import java.util

import com.jxc454.models.SimpleMessages.SimpleInt
import org.apache.kafka.common.serialization.Serializer

class PbIntSerializer extends Serializer[SimpleInt] {
  override def serialize(topic: String, pb: SimpleInt): Array[Byte] = pb.toByteArray

  override def configure(configs: util.Map[String, _], isKey: Boolean): Unit = ()

  override def close(): Unit = ()
}

File: service/src/main/java/com/jxc454/histoservice/PbIntSerializer.java
package com.jxc454.histoservice;

import com.jxc454.models.SimpleMessages.SimpleInt;
import org.apache.kafka.common.serialization.Serializer;

class PbIntSerializer implements Serializer<SimpleInt> {
  public byte[] serialize(String topic, SimpleInt pb) {return pb.toByteArray();}

  public void configure(java.util.Map<String, ?> configs, boolean isKey) {}

  public void close() {}
}
